lib/stack_master/aws_driver/s3.rb

```diff
@@ -0,0 +1,70 @@
+require 'digest/md5'
+
+module StackMaster
+  module AwsDriver
+    class S3ConfigurationError < StandardError; end
+
+    class S3
+      attr_accessor :region, :bucket, :prefix, :files
+
+      def initialize(region: nil, bucket: nil, prefix: nil, files: nil)
```

Comment:
That clarification sort of nullifies my question.  I'll change the implementation, and then the spec tests can largely stay the same.